File: frontend/src/app/core/services/agency-settings.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { environment } from '../../../environments/environment';

export interface AgencySettings {
  id?: string;
  name?: string;
  representativeName?: string;
  address?: string;
  city?: string;
  postalCode?: string;
  email?: string;
  phone?: string;
  website?: string;
  siret?: string;
  professionalCardNumber?: string;
  prefecture?: string;
  guaranteeAmount?: string;
  guaranteeInsurer?: string;
  signatureImagePath?: string;
  logoPath?: string;
}

interface ApiResponse<T> { success: boolean; data: T; }

@Injectable({ providedIn: 'root' })
export class AgencySettingsService {
  private readonly http = inject(HttpClient);
  private readonly apiUrl = `${environment.apiUrl}/agency-settings`;

  get(): Observable<AgencySettings> {
    return this.http.get<ApiResponse<AgencySettings>>(this.apiUrl).pipe(map(r => r.data));
  }

  update(settings: AgencySettings): Observable<AgencySettings> {
    return this.http.put<ApiResponse<AgencySettings>>(this.apiUrl, settings).pipe(map(r => r.data));
  }

  uploadSignature(file: File): Observable<AgencySettings> {
    const form = new FormData();
    form.append('file', file);
    return this.http.post<ApiResponse<AgencySettings>>(`${this.apiUrl}/signature`, form).pipe(map(r => r.data));
  }

  uploadLogo(file: File): Observable<AgencySettings> {
    const form = new FormData();
    form.append('file', file);
    return this.http.post<ApiResponse<AgencySettings>>(`${this.apiUrl}/logo`, form).pipe(map(r => r.data));
  }
}

File: backend/src/main/java/fr/imoblex/modules/agency/service/AgencySettingsService.java
package fr.imoblex.modules.agency.service;

import fr.imoblex.modules.agency.dto.AgencySettingsRequest;
import fr.imoblex.modules.agency.dto.AgencySettingsResponse;
import fr.imoblex.modules.agency.entity.AgencySettings;
import fr.imoblex.modules.agency.repository.AgencySettingsRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
@Transactional
public class AgencySettingsService {

    private final AgencySettingsRepository repository;

    public AgencySettingsResponse get() {
        List<AgencySettings> all = repository.findAll();
        if (all.isEmpty()) {
            AgencySettings defaults = new AgencySettings();
            defaults.setName("IMOBLEX");
            return toResponse(repository.save(defaults));
        }
        return toResponse(all.get(0));
    }

    public AgencySettings getEntity() {
        List<AgencySettings> all = repository.findAll();
        if (all.isEmpty()) {
            AgencySettings defaults = new AgencySettings();
            defaults.setName("IMOBLEX");
            return repository.save(defaults);
        }
        return all.get(0);
    }

    public AgencySettingsResponse update(AgencySettingsRequest req) {
        AgencySettings entity = getEntity();
        if (req.getName() != null) entity.setName(req.getName());
        if (req.getRepresentativeName() != null) entity.setRepresentativeName(req.getRepresentativeName());
        if (req.getAddress() != null) entity.setAddress(req.getAddress());
        if (req.getCity() != null) entity.setCity(req.getCity());
        if (req.getPostalCode() != null) entity.setPostalCode(req.getPostalCode());
        if (req.getEmail() != null) entity.setEmail(req.getEmail());
        if (req.getPhone() != null) entity.setPhone(req.getPhone());
        if (req.getWebsite() != null) entity.setWebsite(req.getWebsite());
        if (req.getSiret() != null) entity.setSiret(req.getSiret());
        if (req.getProfessionalCardNumber() != null) entity.setProfessionalCardNumber(req.getProfessionalCardNumber());
        if (req.getPrefecture() != null) entity.setPrefecture(req.getPrefecture());
        if (req.getGuaranteeAmount() != null) entity.setGuaranteeAmount(req.getGuaranteeAmount());
        if (req.getGuaranteeInsurer() != null) entity.setGuaranteeInsurer(req.getGuaranteeInsurer());
        if (req.getSignatureImagePath() != null) entity.setSignatureImagePath(req.getSignatureImagePath());
        if (req.getLogoPath() != null) entity.setLogoPath(req.getLogoPath());
        return toResponse(repository.save(entity));
    }

    private AgencySettingsResponse toResponse(AgencySettings e) {
        return AgencySettingsResponse.builder()
                .id(e.getId())
                .name(e.getName())
                .representativeName(e.getRepresentativeName())
                .address(e.getAddress())
                .city(e.getCity())
                .postalCode(e.getPostalCode())
                .email(e.getEmail())
                .phone(e.getPhone())
                .website(e.getWebsite())
                .siret(e.getSiret())
                .professionalCardNumber(e.getProfessionalCardNumber())
                .prefecture(e.getPrefecture())
                .guaranteeAmount(e.getGuaranteeAmount())
                .guaranteeInsurer(e.getGuaranteeInsurer())
                .signatureImagePath(e.getSignatureImagePath())
                .logoPath(e.getLogoPath())
                .build();
    }
}
